Gruntfile.js

```diff
@@ -46,17 +46,30 @@ module.exports = (grunt) => {
                 spawn: false
             }
           }
+        },
+
+        env: {
+            dev: {
+                NODE_ENV: 'development'
+            },
+            prod: {
+                NODE_NEV: 'production'
+            },
+            test: {
+                NODE_ENV: 'test'
+            }
         }
     });
 
     grunt.loadNpmTasks('grunt-contrib-jshint');
     grunt.loadNpmTasks('grunt-contrib-watch');
     grunt.loadNpmTasks('grunt-express-server');
     grunt.loadNpmTasks('grunt-mocha-test');
+    grunt.loadNpmTasks('grunt-env');
 
-    grunt.registerTask('test', ['jshint', 'mochaTest']);
-    grunt.registerTask('serve', ['express']);
-    grunt.registerTask('test-watch', ['jshint', 'mochaTest', 'watch']);
-    grunt.registerTask('default', ['jshint', 'express', 'watch']);
+    grunt.registerTask('test', ['env:test', 'jshint', 'express', 'mochaTest']);
+    grunt.registerTask('serve', ['env:dev', 'express']);
+    grunt.registerTask('test-watch', ['env:test', 'jshint', 'express', 'mochaTest', 'watch']);
```

Comment:
Still investigating for a better solution on this.
If we separated the execution of server to become a grunt task, we cannot make use of configurations already set in `server.js`. Example is `mysql` configurations. For every test we are going to create, we need to add the mysql configuration before we can use it. 
Another workaround is to make a function to check whether the environment is for testing, if that is the case, we set `spawn: true` in `watch` task which will spawn child processes per reload.